akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -397,12 +397,13 @@ private[http] object HttpServerBluePrint {
         def onPush(): Unit =
           grab(requestParsingIn) match {
             case r: RequestStart =>
-              openRequests = openRequests.enqueue(r)
+              val rs0 = r.copy(headers = r.headers :+ OnComplete(OnCompleteAccess()))
```

Comment:
This would probably be a little nicer when implemented as an attribute (https://doc.akka.io/docs/akka-http/snapshot/common/http-model.html#attributes)